### PR TITLE
Release web 2023 12 01

### PR DESCRIPTION
**Pull request by naveedaz**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary>Click here to see the details of Step 1, Breaking Changes review</summary>

If you are in purview of Step 1 of the diagram, follow the [Breaking Changes review process].  
**IMPORTANT!** This applies even if you believe your PR was mislabeled, for any reason, including tool failure.
</details>

<details> 
<summary>Click here to see the details of Step 2, ARM review</summary>

See https://aka.ms/azsdk/pr-arm-review.
</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[Breaking Changes review process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by naveedaz** [comment]
The breaking changes are all about the DELETE webapp call, which was reviewed by ARM on the service branch 

[Make PUT/PATCH api async for function on aca](https://github.com/Azure/azure-rest-api-specs/pull/27612)

**Comment by ms-zhenhua** [comment]
There are a bunch of breaking changes reported on this PR. Please get a breaking changes approval from Azure Breaking Changes Reviewers azbreakchangereview@microsoft.com and ARM can review the changes afterwards. Once you have the approval, please remove the "ARMChangesRequested" label from the PR to make the PR visible to the ARM reviewers.

**Comment by naveedaz** [comment]
@ms-zhenhua Please take a look at the service/release branch https://github.com/Azure/azure-rest-api-specs/compare/main...release-web-2023-12-01. All changes were reviewed and approved by ARM.

**Comment by vivekjilla** [comment]
@ms-zhenhua breaking changes are now reverted. Can you please help with ARM Review on this.

File: specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json
Comment (gary-x-li) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json:366]: Why setting it to `null`? The response does have a `nextLink` property.

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/ResourceProvider.json
Comment (gary-x-li) [inline specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/ResourceProvider.json:366]: Why setting it to `null`? The response does have a `nextLink` property.

**Comment by gary-x-li** [comment]
>               "$ref": "./CommonDefinitions.json#/definitions/RemotePrivateEndpointConnectionARMResource"

Is this a breaking change?

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/StaticSites.json:3067 in 0db58f3. [](commit_id = 0db58f35cf175f8b0134e636e51e9bcfbcedc93b, deletion_comment = False)

**Comment by gary-x-li** [comment]
>         "dnsRetryAttemptTimeout": {

It's better to include the time unit in the property name. E.g. `...TimeoutSeconds`.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json:3840 in 0db58f3. [](commit_id = 0db58f35cf175f8b0134e636e51e9bcfbcedc93b, deletion_comment = False)

**Comment by gary-x-li** [comment]
>         "dnsMaxCacheTimeout": {

It's better to include the time unit in the property name. E.g. `...TimeoutSeconds`.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json:3850 in 0db58f3. [](commit_id = 0db58f35cf175f8b0134e636e51e9bcfbcedc93b, deletion_comment = False)

**Comment by gary-x-li** [comment]
>         "dnsLegacySortOrder": {

ARM recommends enums over booleans for future proof APIs. 

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible. 

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json:3855 in 0db58f3. [](commit_id = 0db58f35cf175f8b0134e636e51e9bcfbcedc93b, deletion_comment = False)

**Comment by gary-x-li** [comment]
>         "responses": {

Are PUT response sync or async? If async, make sure you set `x-ms-long-running-operation` and return the `Azure-AsyncOperation` header.  [https://azure.github.io/autorest/extensions/#x-ms-long-running-operation](https://azure.github.io/autorest/extensions/#x-ms-long-running-operation)

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:8734 in 0db58f3. [](commit_id = 0db58f35cf175f8b0134e636e51e9bcfbcedc93b, deletion_comment = False)

**Comment by gary-x-li** [comment]
>         "responses": {

Same question regarding sync vs async.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:18376 in 0db58f3. [](commit_id = 0db58f35cf175f8b0134e636e51e9bcfbcedc93b, deletion_comment = False)

File: specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json
Comment (vivekjilla) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json:366]: Can you please help with this @ha0ch1

**Comment by vivekjilla** [comment]
> > ```
> >           "$ref": "./CommonDefinitions.json#/definitions/RemotePrivateEndpointConnectionARMResource"
> > ```
> 
> Is this a breaking change?
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/StaticSites.json:3067 in [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b). [](commit_id = [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b), deletion_comment = False)

@xingyangli1278 Can you please help with this comment

**Comment by vivekjilla** [comment]
> > ```
> >     "dnsRetryAttemptTimeout": {
> > ```
> 
> It's better to include the time unit in the property name. E.g. `...TimeoutSeconds`.
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json:3840 in [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b). [](commit_id = [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b), deletion_comment = False)

@tbhartia can you please help with this and other comments related to dnsconfig

**Comment by vivekjilla** [comment]
> > ```
> >     "responses": {
> > ```
> 
> Are PUT response sync or async? If async, make sure you set `x-ms-long-running-operation` and return the `Azure-AsyncOperation` header. https://azure.github.io/autorest/extensions/#x-ms-long-running-operation
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:8734 in [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b). [](commit_id = [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b), deletion_comment = False)

@DJagwani Can you please help address the sync vs async comments around sitecontainer APIs

**Comment by DJagwani** [comment]
> > > ```
> > >     "responses": {
> > > ```
> > 
> > 
> > Are PUT response sync or async? If async, make sure you set `x-ms-long-running-operation` and return the `Azure-AsyncOperation` header. https://azure.github.io/autorest/extensions/#x-ms-long-running-operation
> > Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json:8734 in [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b). [](commit_id = [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b), deletion_comment = False)
> 
> @DJagwani Can you please help address the sync vs async comments around sitecontainer APIs

These are sync APIs only.

**Comment by xingyangli1278** [comment]
> > > ```
> > >           "$ref": "./CommonDefinitions.json#/definitions/RemotePrivateEndpointConnectionARMResource"
> > > ```
> > 
> > 
> > Is this a breaking change?
> > Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/StaticSites.json:3067 in [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b). [](commit_id = [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b), deletion_comment = False)
> 
> @xingyangli1278 Can you please help with this comment

@vivekjilla 
This was a "breaking change" but this was a bugfix to change the response body to the correct format. This change has already been approved and merged for all previous api versions for Microsoft.Web.

File: specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json
Comment (ha0ch1) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json:366]: @gary-x-li Thanks for brining this up, I've removed the next link property in the response since this API is not expected to return a nextlink property. 

Here is the PR for the fix, could you please take a look? https://github.com/Azure/azure-rest-api-specs/pull/28696/commits/87a53953594d401f60fd762c2c1bb5779b12ce5d

**Comment by tbhartia** [comment]
> > ```
> >     "dnsRetryAttemptTimeout": {
> > ```
> 
> It's better to include the time unit in the property name. E.g. `...TimeoutSeconds`.
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json:3840 in [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b). [](commit_id = [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b), deletion_comment = False)

These properties have already been rolled-out and we are adding them to the swagger now. Unfortunately, we can't change the name of the properties now as it will be a breaking change but we really appreciate the suggestion and will keep that in mind in future.

**Comment by tbhartia** [comment]
> > ```
> >     "dnsLegacySortOrder": {
> > ```
> 
> ARM recommends enums over booleans for future proof APIs.
> 
> Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.
> 
> A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json:3855 in [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b). [](commit_id = [0db58f3](https://github.com/Azure/azure-rest-api-specs/commit/0db58f35cf175f8b0134e636e51e9bcfbcedc93b), deletion_comment = False)

There is no possibility for this property to have more than 2 values in future as well, so we decided to use boolean here.

File: specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/ResourceProvider.json
Comment (ha0ch1) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/ResourceProvider.json:366]: here's my proposed change: https://github.com/Azure/azure-rest-api-specs/pull/28696/commits

Removed nextlinkname from responds body all together

Comment (ha0ch1) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/ResourceProvider.json:366]: responded in another comment

File: specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json
Comment (jocawtho) [reply specification/web/resource-manager/Microsoft.Web/stable/2023-01-01/ResourceProvider.json]: This will be resolved with a backend change, not a swagger change. This has been reverted.

**Comment by raosuhas** [comment]
All commits have already been reviewed here : 

https://github.com/Azure/azure-rest-api-specs/pull/28268/commits

**Comment by kazrael2119** [comment]
need merge this pr before reivew sdk breaking changes https://github.com/Azure/azure-rest-api-specs/pull/28751

**Comment by jocawtho** [comment]
> need merge this pr before reivew sdk breaking changes #28751

This is complete now.

**Review by rkmanda — APPROVED** [review]

**Pull request merged**
